cpp/mcap/include/mcap/mcap.inl

```diff
@@ -0,0 +1,168 @@
+// Do not compile on systems with non-8-bit bytes
+static_assert(CHAR_BIT == 8);
+
+namespace mcap {
+
+// Public API //////////////////////////////////////////////////////////////////
+
+McapWriter::~McapWriter() {
+  close();
+}
+
+void McapWriter::open(std::ostream& stream, const McapWriterOptions& options) {
+  stream_ = &stream;
+  writeMagic();
+  write(Header{options.profile, options.library, options.metadata});
+}
+
+void McapWriter::close() {
+  if (!stream_) {
+    return;
+  }
+  write(mcap::Footer{0, 0});
+  writeMagic();
+  stream_->flush();
+  stream_ = nullptr;
+}
+
+void McapWriter::registerChannel(mcap::ChannelInfo& info) {
```

Comment:
naming: maybe `addChannel`?

Reply:
At Cruise we recommended in our style guide to use pointers for output parameters. That way the call site is clear, e.g. `registerChannel(&info)`. I would prefer using the same style in this codebase too

Reply:
I remember that guideline, but I thought there was also a guideline to use `&` for `inout` parameters.

Reply:
or like I did in ws-protocol, `ChannelId addChannel(const ChannelInfoWithoutId&)` 🤷